indico/MaKaC/common/cache.py

```diff
@@ -472,7 +482,30 @@ def delete(self, key):
         return 1
 
 
+class MemcachedCacheClient(CacheClient):
+    """Memcached-based cache client"""
+
+    @staticmethod
+    def convert_ttl(ttl):
+        """Convert a ttl in seconds to a timestamp for use with memcached."""
+        return int(time.time()) + ttl if ttl != 0 else 0
+
+    def __init__(self, servers):
+        import memcache
+        self._client = memcache.Client(servers)
+
+    def set(self, key, val, ttl=0):
+        return self._client.set(key, val, self.convert_ttl(ttl))
+
+    def get(self, key):
+        return self._client.get(key)
+
+    def delete(self, key):
+        return self._client.delete(key)
+
+
 class GenericCache(object):
+    """GenericCache wraps the correct CacheClient depending on the setting of the application."""
```

Comment:
I'd go for something more concise like this:

    """A simple cache interface that supports various backends"""